app/src/main/java/com/asma/tasky/feature_authentication/presentation/components/FormTextField.kt

```diff
@@ -0,0 +1,149 @@
+    isPasswordVisible: Boolean = false,
+    onPasswordToggleClick: (Boolean) -> Unit = {},
+    onValueChange: (String) -> Unit
+) {
+
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+
+            .then(modifier)
+    ) {
+        TextField(
+            value = text,
+            onValueChange = {
+                if (it.length <= maxLength) {
+                    onValueChange(it)
+                }
+            },
+            maxLines = maxLines,
+            textStyle = style,
+            colors = TextFieldDefaults.textFieldColors(
+                backgroundColor = backgroundColor,
+                focusedIndicatorColor = Color.Transparent,
+                unfocusedIndicatorColor = Color.Transparent,
+                disabledIndicatorColor = Color.Transparent
+            ),
+            placeholder = {
+                Text(
+                    text = hint,
+                    style = MaterialTheme.typography.body1
+                )
+            },
+            isError = error != "",
+            keyboardOptions = KeyboardOptions(
+                keyboardType = keyboardType
+            ),
+            visualTransformation = if (!isPasswordVisible && isPasswordToggleDisplayed) {
+                PasswordVisualTransformation()
+            } else {
+                VisualTransformation.None
+            },
+            singleLine = singleLine,
+            trailingIcon = if (isPasswordToggleDisplayed) {
+                val icon: @Composable () -> Unit = {
+                    IconButton(
+                        onClick = {
+                            onPasswordToggleClick(!isPasswordVisible)
+                        }
+                    ) {
+                        Icon(
+                            imageVector = if (isPasswordVisible) {
+                                Icons.Filled.VisibilityOff
+                            } else {
+                                Icons.Filled.Visibility
+                            },
+                            tint = Color.LightGray,
+                            contentDescription = if (isPasswordVisible) {
+                                stringResource(id = R.string.visible_password_content_description)
+                            } else {
+                                stringResource(id = R.string.hidden_password_content_description)
+                            }
+                        )
+                    }
+                }
+                icon
+            } else if(isCheckMarkDisplayed){
+                val icon: @Composable () -> Unit = {
+                    IconButton(
+                        onClick = {}
+                    ) {
+                        Icon(imageVector = Icons.Filled.Check,
+                            tint = Green,
+                            contentDescription = "check mark"
+                        )
+                    }
+                }
+                icon
+            }else null,
```

Comment:
Any reason, you extracted the if-statements out of the composable function and preferred to return composable lambdas instead? 🤔